### PR TITLE
Re-enable webusb feature

### PR DESCRIPTION
**Pull request by simonhong**
This api was disabled by https://github.com/brave/brave-core/pull/114

Fix https://github.com/brave/brave-browser/issues/4669

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [x] Public documentation has been updated as necessary. For instance:
  - [x] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [x] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Check `navigator.usb` doesn't return `undefined` in console.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by keepkeyjon** [comment]
Thanks for considering adding WebUSB back! We (and our users) would love to have it enabled in Brave.

What was the rationale for removing it in the first place? From the [thread in #13](https://github.com/brave/brave-browser/issues/13#issuecomment-376991976), it seems it was done out of privacy concerns. Does the need to pair devices in order for websites to talk to them mitigate that?

**Comment by diracdeltas** [comment]
@keepkeyjon it was disabled due to a security issue in https://github.com/brave/browser-laptop/issues/13374.

**Comment by diracdeltas** [comment]
according to https://www.yubico.com/2018/06/webusb-and-responsible-disclosure/, the original security issue was fixed in chromium 67 so we should be ok to re-enable webusb now.

@simonhong please make sure all necessary wiki pages (linked in the PR template) that mention webusb are updated once this is merged. thanks!

**Comment by simonhong** [comment]
@diracdeltas two document are updated.

File: renderer/brave_content_renderer_client.cc [hunk 1]
```diff
@@ -8,8 +9,8 @@
 BraveContentRendererClient::BraveContentRendererClient()
     : ChromeContentRendererClient() {}
 
-void BraveContentRendererClient::SetRuntimeFeaturesDefaultsBeforeBlinkInitialization() {
-  blink::WebRuntimeFeatures::EnableWebUsb(false);
+void BraveContentRendererClient::
+SetRuntimeFeaturesDefaultsBeforeBlinkInitialization() {
   blink::WebRuntimeFeatures::EnableSharedArrayBuffer(false);
```
Comment (bbondy) [inline renderer/brave_content_renderer_client.cc]: Not your issue, but shouldn't we be calling the base class here?


```
void ChromeContentRendererClient::
    SetRuntimeFeaturesDefaultsBeforeBlinkInitialization() {
  // The performance manager service interfaces are provided by the chrome
  // embedder only.
  blink::WebRuntimeFeatures::EnablePerformanceManagerInstrumentation(true);

// Web Share is shipped on Android, experimental otherwise. It is enabled here,
// in chrome/, to avoid it being made available in other clients of content/
// that do not have a Web Share Mojo implementation.
#if defined(OS_ANDROID)
  blink::WebRuntimeFeatures::EnableWebShare(true);
  blink::WebRuntimeFeatures::EnableWebShareV2(true);
#endif

  if (base::FeatureList::IsEnabled(subresource_filter::kAdTagging))
    blink::WebRuntimeFeatures::EnableAdTagging(true);
}
```

**Comment by snyderp** [comment]
Just to surface something that came up in #privacy today

Clearing site data does not currently clear suite permissions to access USB devices.  Whether thats part of this issue, or its own issue, that seems like something important to work out before enabling WebUSB

**Comment by bbondy** [comment]
> Just to surface something that came up in #privacy today
> 
> Clearing site data does not currently clear suite permissions to access USB devices. Whether thats part of this issue, or its own issue, that seems like something important to work out before enabling WebUSB

Please post a different issue for that @snyderp , I think we want to have better webcompat sooner than later.

File: renderer/brave_content_renderer_client.cc (hunk 1 above)
Comment (simonhong) [reply renderer/brave_content_renderer_client.cc]: @bbondy  Yup, we should call base class' method also if we don't want prevent all.
I assume that when @jumde pushed https://github.com/brave/brave-core/pull/114, base class methods would be like below.
```
void ChromeContentRendererClient::
    SetRuntimeFeaturesDefaultsBeforeBlinkInitialization() {
// Web Share is shipped on Android, experimental otherwise. It is enabled here,
// in chrome/, to avoid it being made available in other clients of content/
// that do not have a Web Share Mojo implementation.
#if defined(OS_ANDROID)
  blink::WebRuntimeFeatures::EnableWebShare(true);
#endif
}
```
So, @pj might be skipped calling it.
Added.

**Comment by simonhong** [comment]
Only unrelated one step in CI was failed - `Copy file to S3 - (68 ms in self)`

File: renderer/brave_content_renderer_client.cc (hunk 1 above)
Comment (bbondy) [reply renderer/brave_content_renderer_client.cc]: I think maybe it is safest to always call it and just undo anything we don't want to do. But probably there is nothing else we don't want to do.  We'll have android support soon btw.

**Comment by simonhong** [comment]
Passed in CI. Any more comments on this?

**Comment by kjozwiak** [comment]
Went through the following verifications:

* ensured that `navigator.usb` isn't returning `undefined` in DevTools
* connect to a `Ledger Nano S` using https://mycrypto.com/
* went through a `Trezor` setup using https://connect.trezor.io without any issues
* connected to `Trezo`r using https://mycrypto.com/
* @mrose17 confirmed that he can connect to his `KeepKey` using `0.68.36 Chromium: 75.0.3770.87`

Examples:

Screen Shot 2019-06-11 at 6 23 44 PM

Screen Shot 2019-06-11 at 7 23 18 PM

**Comment by kjozwiak** [comment]
> Just to surface something that came up in #privacy today
> 
> Clearing site data does not currently clear suite permissions to access USB devices. Whether thats part of this issue, or its own issue, that seems like something important to work out before enabling WebUSB

@snyderp did you end up creating an issue for the above? Just double checking to make sure it doesn't end up falling through the cracks.

**Comment by snyderp** [comment]
@kjozwiak the issue was that I'm a dingbat and was confused by the "Clear data" dialog.  I didn't realize the tabs on the "clear data" dialog control the behavior of the "clear" button at the bottom (instead of just showing more information about the same action).

TL;DR: i'm a dumb-dumb and no issue needed.  I appreciate you double checking though :)

**Comment by kjozwiak** [comment]
@snyderp haha no worries! Appreciate the update 👍
